configuration/config2/EnvironmentNode.cpp

```diff
@@ -73,138 +73,166 @@ std::map<std::string, std::vector<std::shared_ptr<EnvironmentNode>>> Environment
 }
 
 
-std::vector<std::shared_ptr<EnvValue>> EnvironmentNode::getAttributes() const
+std::shared_ptr<EnvironmentNode> EnvironmentNode::getParent() const
 {
-	std::vector<std::shared_ptr<EnvValue>> attributes;
-
-	for (auto attrIt = m_attributes.begin(); attrIt != m_attributes.end(); ++attrIt)
+	std::shared_ptr<EnvironmentNode> pParent;
+	if (!m_pParent.expired())
 	{
-		attributes.push_back(attrIt->second);
+		pParent = m_pParent.lock();
 	}
-	return attributes;
+	return pParent;
+}
+
+
+std::vector<std::shared_ptr<EnvValue>> EnvironmentNode::getAttributes() const
+{
+    std::vector<std::shared_ptr<EnvValue>> attributes;
+
+    for (auto attrIt = m_attributes.begin(); attrIt != m_attributes.end(); ++attrIt)
+    {
+        attributes.push_back(attrIt->second);
+    }
+    return attributes;
+}
+
+
+void EnvironmentNode::setAttributeValues(const std::vector<ValueDef> &values, Status &status, bool allowInvalid, bool forceCreate)
+{
+    for (auto it = values.begin(); it != values.end(); ++it)
+    {
+        setAttributeValue((*it).name, (*it).value, status, allowInvalid, forceCreate);
+    }
 }
 
 
 // should probably return a status object, and put path/valueName in there
-bool EnvironmentNode::setAttributeValue(const std::string &name, const std::string &value, bool allowInvalid, bool forceCreate)
+void EnvironmentNode::setAttributeValue(const std::string &attrName, const std::string &value, Status &status, bool allowInvalid, bool forceCreate)
```

Comment:
then let's do so... "// should probably return a status object, and put path/valueName in there"